distribution/lib/Standard/AWS/0.0.0-dev/src/S3/S3_File.enso

```diff
@@ -182,17 +197,37 @@ type S3_File
 
     ## ICON data_output
        Deletes the object.
-    delete : Nothing
-    delete self = if self.is_directory then Error.throw (S3_Error.Error "Deleting S3 folders is currently not implemented." self.uri) else
-        if self.exists.not then Error.throw (File_Error.Not_Found self) else
-            self.delete_if_exists
+
+       Arguments:
+       - recursive: If the target is a non-empty directory, it will only be
+         removed if this is set to `True`. Defaults to `False`, meaning that the
+         operation will fail if the directory is not empty. This option has no
+         effect for files or data links.
+    delete : Boolean -> Nothing
+    delete self (recursive : Boolean = False) =
+        # TODO improve recursive deletion for S3 folders: https://github.com/enso-org/enso/issues/9704
+        if self.is_directory then Error.throw (S3_Error.Error "Deleting S3 folders is currently not implemented." self.uri) else
```

Comment:
It looks like S3 folder deletion is implemented in `delete_if_exists`, why is it different here?

Reply:
Good point, I removed that and rephrased the TODO